app/parsers/bulkrax/application_parser.rb

```diff
@@ -37,15 +38,94 @@ def records(opts = {})
       raise 'must be defined'
     end
 
-    def import_fields
-      raise 'must be defined'
+    def import_file_path
+      @import_file_path ||= self.parser_fields['import_file_path']
+    end
+
+    def create_collections
+      raise 'must be defined' if importer?
+    end
+
+    def create_works
+      raise 'must be defined' if importer?
+    end
+
+    # Optional, only used by certain parsers
+    # Other parsers should override with a custom or empty method
+    # Will be skipped unless the record is a Hash
+    def create_parent_child_relationships
+      parents = setup_parents
+      return if parents.blank?
+
+      parents.each do | key, value |
+        parent = entry_class.where(
+          identifier: key,
+          importerexporter_id: importerexporter.id,
+          importerexporter_type: 'Bulkrax::Importer',
+        ).first
+
+        # not finding the entries here indicates that the given identifiers are incorrect
+        # in that case we should log that
+        children = []
+        children = value.map do | child |
+          entry_class.where(
+            identifier: child,
+            importerexporter_id: importerexporter.id,
+            importerexporter_type: 'Bulkrax::Importer',
+          ).first
```

Comment:
children will potentially contain `[Child, nil, Child]`

Reply:
and if only 1 is nil, the count will be the same below

Reply:
I think you need `child = entry_class.where...` and then deal with incrementing the error here where you know what it is.  Probably coupled with replacing map with each to build the children array as you go.

Reply:
does the additon of `.compact.uniq` solve this? (as per latest)

Reply:
oh, I see, the compact does remove the nils, so yes, that seems ok.